api/spec/requests/v1/payments_controller_spec.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe V1::PaymentsController, type: :request do
+  describe '#generate_transaction' do
+    subject(:transaction_adder) { instance_double(SEPA::CreditTransfer, add_transaction: true) }
+
+    let(:payments_controller) { V1::PaymentsController.new }
+    let(:expense_sheet) { create :expense_sheet, :ready_for_payment }
+    let(:user) { expense_sheet.user }
+
+    before do
+      allow(SEPA::CreditTransfer).to receive(:new).and_return transaction_adder
+
+      payments_controller.send(:generate_pain, [expense_sheet])
```

Comment:
Usually, the `send` shouldn't be used in specs, if you're calling private methods. This seems more like you would need a `PAINGenerationService` which has a public generate method and test this directly.